apps/src/studio/studio.js

```diff
@@ -588,6 +589,25 @@ function callHandler (name, allowQueueExtension) {
 }
 
 /**
+ * Performs movement on a list of Projectiles or Items. Removes items from the
+ * list automatically when they move out of bounds
+ */
+function performItemOrProjectileMoves (list) {
+  for (var i = 0; i < list.length; i++) {
+    list[i].moveToNextPosition();
+    if (list[i].outOfBounds()) {
+      list[i].removeElement();
+      list.splice(i, 1);
+      // decrement i because we just removed an item from the array. We want to
+      // keep i as the same value for the next iteration through this loop
```

Comment:
I think an alternate strategy here might be to walk backwards instead of forwards through the list, so that if you remove the current item, i-- will just go to the next item anyway.